idaes/surrogate/pysmo/tests/test_sampling_modified.py

```diff
@@ -297,35 +301,53 @@ def setUp(self):
     test_prime_base_to_decimal_02: Test behaviour with 0.01 in base 20 to base 10
     test_prime_base_to_decimal_03: working with base 1
     test_prime_base_to_decimal_04: working with base -1
-    
-    test_data_sequencing: in the notes, (3,2) -> expected [0, 0.5, 0.75], but returns [0.  , 0.5 , 0.25]
+
+    test_selection_columns_preprocessing_01: Test behaviour with no labels specified
+    test_selection_columns_preprocessing_02: Test behaviour with only xlabels specified with multiple elements, same order as in dataframe
+    test_selection_columns_preprocessing_03: Test behaviour with only xlabels specified with multiple elements, different order from dataframe
+    test_selection_columns_preprocessing_04: Test behaviour with only xlabels specified with only 1 element
+    test_selection_columns_preprocessing_05: Test behaviour with only ylabels specified with multiple elements, same order as in dataframe
+    test_selection_columns_preprocessing_06: Test behaviour with only ylabels specified with multiple elements, different order from dataframe
+    test_selection_columns_preprocessing_07: Test behaviour with only ylabels specified with only 1 element
+    test_selection_columns_preprocessing_11: Test behaviour with both xlabels and ylabels specified; multiple x and single y
+    test_selection_columns_preprocessing_08: Test behaviour with both xlabels and ylabels specified; single x and single y; ordered
+    test_selection_columns_preprocessing_09: Test behaviour with both xlabels and ylabels specified; single x and multiple y
+    test_selection_columns_preprocessing_10: Test behaviour with both xlabels and ylabels specified; single x and single y; unordered
+    test_selection_columns_preprocessing_12: Test behaviour with both xlabels and ylabels specified; multiple x and y, all initial columns present
+    test_selection_columns_preprocessing_13: Test behaviour when non-existent column name is supplied in label - should raias ValueError
+    test_selection_columns_preprocessing_14 - 26: Repeats above tests 1-13 for case where input is a numpy array rather than a dataframe
```

Comment:
I love this and I think it's an excellent way to document what the various tests do. A few thoughts

- If tests `14` through `26` are the same as `01` through `13` except for one value, maybe we should consider if they could be parametrized using either `pytest.mark.parametrize()` or a parametrized fixture
  - Aside for reducing repetition in general, this would avoid having to rename 13 functions in case additional tests are added in the future
 - Related to the previous point, but it might make sense to move the information about the purpose of each test closer to the implementation, e.g. in the test function's docstring (or, in the name of the function itself)

Reply:
I'm not sure how using `parametrize() ` would work. Depending on the data type, the expected results and thus the checks are different.

Reply:
The comments have been moved to the docstrings of each test.